king_phisher/client/dialogs/configuration.py

```diff
@@ -274,6 +290,22 @@ def interact(self):
 		self.dialog.destroy()
 		return response
 
+	def save_proxy_settings(self):
+		proxy_url = urllib.parse.urlparse(self.gtk_builder_get('entry_proxy_url').get_text())
+		proxy_username = self.gtk_builder_get('entry_proxy_username').get_text()
+		proxy_password = self.gtk_builder_get('entry_proxy_password').get_text()
+		if proxy_url.hostname and not proxy_url.scheme or proxy_url.scheme and not proxy_url.hostname:
```

Comment:
Simplify again with `if proxy_url and not (proxy_url.hostname and proxy_url.scheme)`. This would evaluate to true when anything is set in the proxy_url but the hostname or scheme is missing.